pkg/controller/build/build_controller.go

```diff
@@ -0,0 +1,1043 @@
+baseurl = https://cdn.redhat.com/content/dist/rhel8/8/x86_64/baseos/os
+enabled = 1
+gpgcheck = 0
+sslverify = 0
+sslclientkey = /etc/pki/entitlement/entitlement-key.pem
+sslclientcert = /etc/pki/entitlement/entitlement.pem
+metadata_expire = 86400
+enabled_metadata = 1
+
+[rhel-8-for-x86_64-appstream-rpms]
+name = Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)
+baseurl = https://cdn.redhat.com/content/dist/rhel8/8/x86_64/appstream/os
+enabled = 1
+gpgcheck = 0
+sslverify = 0
+sslclientkey = /etc/pki/entitlement/entitlement-key.pem
+sslclientcert = /etc/pki/entitlement/entitlement.pem
+metadata_expire = 86400
+enabled_metadata = 1
+`
+
+		// Make an ignition to stuff into our machineconfig
+		ignConfig := ctrlcommon.NewIgnConfig()
+		ignConfig.Storage.Files = append(ignConfig.Storage.Files, NewIgnFile("/etc/yum.repos.d/redhat.repo", redhatRepo))
+
+		// Get our entitlement secrets out of the managed namespace
+		entitlements, err := ctrl.kubeclient.CoreV1().Secrets("openshift-config-managed").Get(context.TODO(), "etc-pki-entitlement", metav1.GetOptions{})
+		if err != nil {
+			glog.Warningf("Could not retrieve entitlement secret: %s", err)
+			return
+		}
+
+		// Add the key to the file list
+		if key, ok := entitlements.Data["entitlement-key.pem"]; ok {
+			ignConfig.Storage.Files = append(ignConfig.Storage.Files, NewIgnFile("/etc/pki/entitlement/entitlement-key.pem", string(key)))
+		}
+
+		// Add the public key to the file list
+		if pub, ok := entitlements.Data["entitlement.pem"]; ok {
+			ignConfig.Storage.Files = append(ignConfig.Storage.Files, NewIgnFile("/etc/pki/entitlement/entitlement.pem", string(pub)))
+		}
+
+		// Now it's a machineconfig
+		mc, err := ctrlcommon.MachineConfigFromIgnConfig(pool.Name, entitledConfigName, ignConfig)
+		if err != nil {
+			glog.Warningf("Could not create machineconfig for entitlements: %s", err)
+		}
+
+		// Add it to the list for this pool
+		_, err = ctrl.client.MachineconfigurationV1().MachineConfigs().Create(context.TODO(), mc, metav1.CreateOptions{})
+		if err != nil {
+			glog.Warningf("Failed to add entitlements to layered pool: %s", err)
+		}
+	}
+
+}
+
+// annotatePoolWithNewestImage looks in the corresponding image stream for a pool and annotates the name of the image, and it's
+// corresponding rendered-config, which it retrieves from the image's docker metadata labels that we added during our build
+func (ctrl *Controller) annotatePoolWithNewestImage(imageStream *imagev1.ImageStream, pool *mcfgv1.MachineConfigPool) error {
+
+	// We don't want to crash if these are empty
+	if pool.Annotations == nil {
+		pool.Annotations = map[string]string{}
+	}
+
+	// Grab the latest tag from the imagestream. If we don't have one, nothing happens
+	for _, tag := range imageStream.Status.Tags {
+		if len(tag.Items) > 0 {
+
+			// I might have an older image that has right machine config content, but some
+			// other content might have changed (like, I dunno, base image) so we shouldn't go back
```

Comment:
I think some of my other comments were based on the assumption that the base image won't have changed. Hmm may have to think about this more on Monday :thinking: